src/main/java/uk/gov/ons/registers/cdap/service/CHService.java

```diff
@@ -0,0 +1,131 @@
+package uk.gov.ons.registers.cdap.service;
+
+import co.cask.cdap.api.annotation.UseDataSet;
+import co.cask.cdap.api.dataset.table.Get;
+import co.cask.cdap.api.dataset.table.Row;
+import co.cask.cdap.api.dataset.table.Scanner;
+import co.cask.cdap.api.dataset.table.Table;
+import co.cask.cdap.api.service.AbstractService;
+import co.cask.cdap.api.service.http.AbstractHttpServiceHandler;
+import co.cask.cdap.api.service.http.HttpServiceRequest;
+import co.cask.cdap.api.service.http.HttpServiceResponder;
+import com.google.gson.Gson;
+import com.google.gson.JsonElement;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import java.net.HttpURLConnection;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Map;
+
+public class CHService extends AbstractService {
+
+    static final String SERVICE_NAME = "CHService";
+    private static final String SERVICE_DESC = "Service that returns A JSON object of company data based on CompanyNumber";
+    private static final String POSTCODE_COLUMN = "regaddress_postcode";
+
+    @Override
+    protected void configure() {
+        setName(SERVICE_NAME);
+        setDescription(SERVICE_DESC);
+        addHandler(new CHdataHandler());
+    }
+
+    /**
+     * Handler which defines HTTP endpoints to access information stored in the
+     * { @number CHdata} Dataset.
+     */
+    public static class CHdataHandler extends AbstractHttpServiceHandler {
+        private static final Logger LOG = LoggerFactory.getLogger(CHdataHandler.class);
```

Comment:
Retrieving a logger like so is good, but can be made more robust (to cut-and-paste) by using the following:
`
private static final Logger LOGGER = LoggerFactory.getLogger(MethodHandles.lookup().lookupClass());
`
(as recommended on https://www.slf4j.org/faq.html#declaration_pattern)

For bonus points, if you are going to be doing this a lot, then it might be worth modifying the template used by Intellij (assuming that's what you are using) and add the Logger statement in there.  To do so :   Settings -> File and Code Templates -> Class:

![2018-06-19_14-32-50](https://user-images.githubusercontent.com/1554671/41600371-aee27b4a-73cd-11e8-8496-b68cf5e2eaca.png)